### PR TITLE
Serialization: Serialize both the substituted and unsubstituted type for a NameAliasType

### PR DESCRIPTION
**Pull request by slavapestov**

### CODE REVIEW

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

File: validation-test/compiler_scale/bind_nested_extension_decl.gyb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+// RUN: %scale-test --sum-multi --typecheck --begin 5 --end 16 --step 5 --select NumDeclsValidated %s
+// REQUIRES: OS=macosx
```
Comment (jrose-apple) [inline validation-test/compiler_scale/bind_nested_extension_decl.gyb]: Why OS=macosx?

**Review by DougGregor — APPROVED** [review]
LGTM, thanks!

File: validation-test/compiler_scale/bind_nested_extension_decl.gyb (hunk 1 above)
Comment (slavapestov) [reply validation-test/compiler_scale/bind_nested_extension_decl.gyb]: All the scale tests have that...

**Pull request merged**